Sources/BraintreeThreeDSecure/BTThreeDSecureClient.swift

```diff
@@ -4,49 +4,112 @@ import Foundation
+    
+    // MARK: - Public Methods
+    
+    /// Starts the 3DS flow using a BTThreeDSecureRequest.
+    /// - Parameters:
+    ///   - request: A BTThreeDSecureRequest request.
+    ///   - completionBlock: This completion will be invoked exactly once when the 3DS flow is complete or an error occurs.
+    public func startPaymentFlow(_ request: BTThreeDSecureRequest, completion: @escaping (BTThreeDSecureResult?, Error?) -> Void) {
+        apiClient.sendAnalyticsEvent("ios.three-d-secure.start-payment.selected")
+        
+        self.request = request
+        self.merchantCompletion = completion
+        
+        apiClient.sendAnalyticsEvent("ios.three-d-secure.initialized")
+
+        apiClient.fetchOrReturnRemoteConfiguration { [weak self] configuration, error in
+            guard let self else { return }
 
-/// Extension on BTPaymentFlowClient for 3D Secure
-extension BTPaymentFlowClient {
+            if let error {
+                completion(nil, error)
+                return
+            }
 
-    // MARK: - Public Methods
+            guard let configuration, configuration.cardinalAuthenticationJWT != nil else {
+                NSLog("%@ Missing the required Cardinal authentication JWT.", BTLogLevelDescription.string(for: .critical))
+                let error = BTThreeDSecureError.configuration("Missing the required Cardinal authentication JWT.")
+                completion(nil, error)
+                return
+            }
+
+            if self.request?.amount?.decimalValue.isNaN == true || self.request?.amount == nil {
+                NSLog("%@ BTThreeDSecureRequest amount can not be nil or NaN.", BTLogLevelDescription.string(for: .critical))
+                let error = BTThreeDSecureError.configuration("BTThreeDSecureRequest amount can not be nil or NaN.")
+                completion(nil, error)
+                return
+            }
+
+            if self.request?.threeDSecureRequestDelegate == nil {
+                let error = BTThreeDSecureError.configuration("Configuration Error: threeDSecureRequestDelegate can not be nil when versionRequested is 2.")
+                completion(nil, error)
+                return
+            }
 
+            self.prepareLookup(request: request) { error in
+                if let error {
+                    self.apiClient.sendAnalyticsEvent("ios.three-d-secure.start-payment.failed")
+                    completion(nil, error)
+                    return
+                }
+
+                self.start(request: request, configuration: configuration)
+            }
+        }
+    }
+        
     /// Creates a stringified JSON object containing the information necessary to perform a lookup.
     /// - Parameters:
     ///   - request: The `BTPaymentFlowRequest` object where prepareLookup was called.
     ///   - completion: This completion will be invoked exactly once with the client payload string or an error.
     @objc(prepareLookup:completion:)
     public func prepareLookup(
```

Comment:
why would merchants call this function explicitly?

Reply:
Good question - this is exposed for ❄️ special merchants who perform server-side 3DS